test/spec/spec.model.js

```diff
@@ -240,6 +241,22 @@ describe('Model model', function () {
             model.url.should.have.been.calledWithExactly({ url: 'foo' });
         });
 
+        it('can handle a parsed URL object', function () {
+            var url = {
+                protocol: 'http:',
+                port: '1234',
+                hostname: 'proxy-example.com',
+                path: '/'
+            };
+            model.url = sinon.stub().returns(url);
```

Comment:
In an ideal world I'd have thought it would be nice to avoid the stub here and pass the url in through the options argument to the constructor. But that might cause too much pain in refactoring the tests.